CONTRIBUTING.md

```diff
@@ -4,17 +4,42 @@ Hi there! We're thrilled that you'd like to contribute to this project. Your hel
 
 The majority of contributions won't need to touch any Ruby code at all.
 
+## Getting started
+
+Before you can start contributing to Linguist, you'll need to set up your environment first. Clone the repo and run `script/bootstrap` to install its dependencies.
+
+    git clone https://github.com/github/linguist.git
+    cd linguist/
+    script/bootstrap
+
+To run Linguist from the cloned repository, you will need to generate the code samples first:
+
+    bundle exec rake samples
+
+Run this command each time a [sample][samples] has been modified.
+
+To run Linguist from the cloned repository:
+
+    bundle exec bin/linguist --breakdown
+
+### Dependencies
+
+Linguist uses the [`charlock_holmes`](https://github.com/brianmario/charlock_holmes) character encoding detection library which in turn uses [ICU](http://site.icu-project.org/), and the libgit2 bindings for Ruby provided by [`rugged`](https://github.com/libgit2/rugged). These components have their own dependencies - `icu4c`, and `cmake` and `pkg-config` respectively - which you may need to install before you can install Linguist.
+
+For example, on macOS with [Homebrew](http://brew.sh/): `brew install cmake pkg-config icu4c` and on Ubuntu: `apt-get install cmake pkg-config libicu-dev`.
+
 ## Adding an extension to a language
 
 We try only to add new extensions once they have some usage on GitHub. In most cases we prefer that extensions be in use in hundreds of repositories before supporting them in Linguist.
 
 To add support for a new extension:
 
-1. Add your extension to the language entry in [`languages.yml`][languages], keeping the extensions in alphabetical order.
-1. Add at least one sample for your extension to the [samples directory][samples] in the correct subdirectory.
-1. Open a pull request, linking to a [GitHub search result](https://github.com/search?utf8=%E2%9C%93&q=extension%3Aboot+NOT+nothack&type=Code&ref=searchresults) showing in-the-wild usage.
+1. Add your extension to the language entry in [`languages.yml`][languages], keeping the extensions in alphabetical order, with the exception of the primary extension; the primary extension should be first.
+1. Add at least one sample for your extension to the [samples directory][samples] in the correct subdirectory. We'd prefer this sample to be an example of real-world code showing common usage. The more representative of the structure of the language, the better.
```

Comment:
*"We'd prefer this sample to be an example of"* is redundant and can be reduced to *"We prefer examples of real-world code…"*. The preceding sentence already established context (linguists call it a "topicaliser", I believe).